File: main-website/models/card_model.py
from dataclasses import dataclass
from typing import Optional, Dict, Any

@dataclass
class CardModel:
    image: Optional[str] = None
    title: str = 'No Title'
    description: str = 'No Description'

    @staticmethod
    def from_json(json: Dict[str, Any]) -> 'CardModel':
        return CardModel(
            image=json.get('image', 'No Image'),
            title=json.get('title', 'No Title'),
            description=json.get('description', 'No Description')
        )

File: admin-panel/lib/core/models/card_model.dart
class CardModel {
  String? id;
  String? image;
  String? title;
  String? description;

  CardModel(
      {this.id,
      this.image,
      this.title,
      this.description});

  factory CardModel.fromJson(Map<String, dynamic> json) => CardModel(
        id: json['id'] ?? 'No ID',
        image: json['image'] ?? 'No Image',
        title: json['title'] ?? 'No Title',
        description: json['description'] ?? 'No Description',
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'image': image,
        'title': title,
        'description': description,
      };

  CardModel copyWith({
    String? id,
    String? image,
    String? title,
    String? description,
  }) =>
      CardModel(
          id: id ?? this.id,
          image: image ?? this.image,
          title: title ?? this.title,
          description: description ?? this.description);
}
